src/target/riscv/riscv.c

```diff
@@ -506,6 +506,17 @@ static int find_trigger(struct target *target, int type, bool chained, int *idx)
 	return ERROR_FAIL;
 }
 
+static int find_trigger_by_id(struct target *target, int unique_id)
+{
+	RISCV_INFO(r);
+
+	for (unsigned i = 0; i < r->trigger_count; i++) {
+		if (r->trigger_unique_id[i] == unique_id)
+			return i;
+	}
+	return -1;
+}
```

Comment:
Minor thing to note - In some cases, there will be multiple triggers with the with the same `unique_id`. For example here: https://github.com/riscv/riscv-openocd/pull/777/files#diff-b4aa16f9e42cb8f0934baa7c8e0ec9c70a369bef98b99b26ae2e896c8aa95d6aR766

This function would find just the first of them. This is not an issue for itrigger but would be insufficient for other trigger types.

To avoid possible future mistakes, you may make the function specific for itriggers: `find_itrigger(target)`, if you'd like.

Reply:
Good point. I don't want to limit it to itrigger(). I wrote this function because it will be helpful for etrigger and icount as well.

How about find_first_trigger_by_id()?

Reply:
That sounds fine (and very clear).